app/assets/javascripts/check-for-edits-before-submit.js

```diff
@@ -0,0 +1,18 @@
+(function ($) {
+  'use strict';
+
+  var form = $('.js-check-for-edits-before-submit');
+  var form_original_data = form.serialize();
+
+  $(".js-before-submit").on('click submit', function() {
+    if (form.serialize() != form_original_data) {
+      if(!confirm('You have unsaved changes to the current month.\n\nProceed and discard changed?')) {
+        return false;
+      }
+    }
+  });
+
+  $('.js-month').on('change', function() {
+    this.closest('form').submit();
```

Comment:
Would be better to target this with a class just in case something else gets involved here, who knows what form we're submitting!